test_generate_timesteps.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+import numpy as np
+import sys 
+sys.path.insert(0, './src')
+from generate_timesteps import generate_timesteps
+
+
+def test_generate_mesh():
+    simulation_length = 10
+    num_steps = 11
+    expected = np.array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9., 10.])
+    assert np.all(expected == generate_timesteps(simulation_length, num_steps))
+
+
+def test_generate_mesh_2():
+    simulation_length = 15.7
+    num_steps = 20
+    expected = np.zeros(num_steps)
```

Comment:
I think the reason this test is failing is because np.all tests all 64 digits of each entry in the output array. To get less significant figures, use np.allclose. 

Try this:
`for i in range(len(expected)):`
     
    `expected[i] = simulation_length/(num_steps - 1)*i

`assert np.all(expected, generate_timesteps(simulation_length, num_steps), rtol = 1e-8)`

This set of code tests it to 8 significant figures, but can be changed by varying the `rtol` input.

Reply:
You can also import `pytest` and use `pytest.approx()`